### PR TITLE
Auto Name Fetching and Other Tweaks

### PR DESCRIPTION
**Pull request by Kneckter**
This PR includes the following:

- Added functionality to get park names from today's OSM data.
-- Only updates parks with unknown names, in case you already named parks.
-- It saves the name data file and updates the CSV.
- Fixed the timer for getting OSM data.
- Fixed the warning about the truncated double value.
- Updated the bash scrip to have the shebang at the top.
- Changed the nest JSON print to only show with the verbose option.
- Fixed/changed some of the prints. 
-- The progress bars were overwriting eachother even though they were different loops.
-- The reasons output at the end were hard to follow

The data used for the names is pulled from zero hour of the day you run the script. This allows you to limit the OSM requests if you have to rerun the script in the same day. That means if one of your park names was updated today, run the script tomorrow (or let it run on the cron job if you can wait). I also added a file delete so it removes old name files because they can take up a lot of space. Any names you put into the CSV or JSON file in the past will not be lost. This will only update parks that have the DEFAULT_PARK_NAME. 

Adding the shebang to the script should allow people to cron job it easier.

Edit1: Added the `OSM_DATE_4NAME` config option to configure the date used for the name file. Acceptable values are `NONE`, `TODAY`, or a date string like `2019-02-24T00:00:00Z`.

### CODE REVIEW

**Comment by kamieniarz** [comment]
I like the idea but I think the date should be configurable and we should have an option to make it static. Why? I have miltiple custom areas and OSM gived me timeouts when I try to download OSM data for less than half of all my areas so this PR breaks script for me. If I could set a date in config (eg. 10.06.2020) that would be just perfect - osm data downloaded once and I'm ready to go instead of downloading it each day.

**Comment by Kneckter** [comment]
> I like the idea but I think the date should be configurable and we should have an option to make it static. Why? I have miltiple custom areas and OSM gived me timeouts when I try to download OSM data for less than half of all my areas so this PR breaks script for me. If I could set a date in config (eg. 10.06.2020) that would be just perfect - osm data downloaded once and I'm ready to go instead of downloading it each day.

You can now configure the date with a static date or the dynamic TODAY value. NONE is used to disable it.
